packages/core/src/presentation/Presentation.ts

```diff
@@ -194,5 +235,274 @@ export function create(
+        result.results = [{ verified, proof }]
+      } catch (proofError) {
+        const errorStr = String(proofError)
+        result.results = [{ verified: false, error: [errorStr], proof }]
+        errors.push(errorStr)
+      }
+
+      result.verified = result.results.every(({ verified }) => verified)
+    }
+    if (errors.length === 0) {
+      result.error = undefined
+    } else {
+      result.error = errors
+      result.verified = false
+    }
+  } catch (e) {
+    errors.push(String(e))
+    result.verified = false
+    result.error = errors
+  }
+  return result
+}
+
+async function verifyCredential(
+  credential: VerifiableCredential
+): Promise<VerifyCredentialResult> {
+  let proof
+  try {
+    proof = getProof(credential)
+    await KiltAttestationProofV1.verify(
+      credential as KiltCredentialV1,
+      proof as Types.KiltAttestationProofV1
+    )
+    return {
+      verified: true,
+      credential,
+      results: [{ verified: true, proof }],
+    }
+  } catch (error) {
+    return {
+      verified: false,
+      credential,
+      error: [String(error)],
+      results: proof
+        ? [{ verified: false, error: [String(error)], proof }]
+        : [],
+    }
+  }
+}
+
+async function checkStatus(
+  credential: VerifiableCredential
+): Promise<VerificationResult> {
+  try {
+    await KiltRevocationStatusV1.check(credential as KiltCredentialV1)
+    return { verified: true }
+  } catch (e) {
+    return { verified: false, error: [String(e)] }
+  }
+}
+
+/**
+ * Verifies a given Verifiable Presentation and its associated Verifiable Credentials.
+ *
+ * This function:
+ * - Verifies the integrity of the presentation by verifying the embedded data integrity proofs.
+ * - If the presentation is valid, verifies each associated credential.
+ * - Checks the status of each verified credential.
+ * - Returns a composite verification result for the presentation and each credential.
+ *
+ * @param presentation - The Verifiable Presentation to be verified.
+ * @param options - Verification options.
+ * @param options.now - The reference time for verification as Date (default is current time).
+ * @param options.challenge - The expected challenge value for the presentation, if any.
+ * @param options.cryptosuites - Array of cryptographic suites to use during verification (default includes suites for `sr25519-jcs-2023`, `eddsa-jcs-2022`, and `es256k-jcs-2023`).
+ * @param options.verifier - The expected verifier for the presentation, if any. This is set as the proof `domain` as well.
+ * @param options.tolerance - The allowed time drift in milliseconds for time-sensitive checks (default is 0).
+ *
+ * @returns An object representing the verification results of the presentation and each associated credential.
+ */
+export async function verify(
+  presentation: VerifiablePresentation,
+  {
+    now = new Date(),
+    tolerance = 0,
+    cryptosuites = [sr25519Suite, eddsaSuite, ecdsaSuite],
+    challenge,
+    verifier,
```

Comment:
This should be called `domain`, since the `verifier` is part of the presentation already.

Reply:
But this just asserts the expected verifier, the domain would be an additional parameter

Reply:
Yeah sorry, swapped those around.